src/k8s/pkg/snap/snap.go

```diff
@@ -79,6 +79,10 @@ func (s *snap) Strict() bool {
 	return meta.Confinement == "strict"
 }
 
+func (s *snap) OnLXD(ctx context.Context) bool {
+	return utils.RunCommand(ctx, "grep", "-qa", "container=lxc", "/proc/1/environ") == nil
```

Comment:
You might want to wait and rebase after https://github.com/canonical/k8s-snap/pull/160/files#diff-e8e4848fe5de62c067838047b54f6c6f0ffacb00fc058e00fcf5168ebd9ee625R21